core/commands/object.go

```diff
@@ -348,6 +355,186 @@ Data should be in the format specified by the --inputenc flag.
 	Type: Object{},
 }
 
+var objectNewCmd = &cmds.Command{
+	Helptext: cmds.HelpText{
+		Tagline: "creates a new object from an ipfs template",
+		ShortDescription: `
+'ipfs object new' is a plumbing command for creating new DAG nodes.
+`,
+		LongDescription: `
+'ipfs object new' is a plumbing command for creating new DAG nodes.
+By default it creates and returns a new empty merkledag node, but
+you may pass an optional template argument to create a preformatted
+node.
+`,
+	},
+	Arguments: []cmds.Argument{
+		cmds.StringArg("template", false, false, "optional template to use"),
+	},
+	Run: func(req cmds.Request, res cmds.Response) {
+		n, err := req.Context().GetNode()
+		if err != nil {
+			res.SetError(err, cmds.ErrNormal)
+			return
+		}
+
+		node := new(dag.Node)
+		if len(req.Arguments()) == 1 {
+			var err error
+			node, err = nodeFromTemplate(req.Arguments()[0])
+			if err != nil {
+				res.SetError(err, cmds.ErrNormal)
+				return
+			}
+		}
+
+		k, err := n.DAG.Add(node)
+		if err != nil {
+			res.SetError(err, cmds.ErrNormal)
+			return
+		}
+		res.SetOutput(&Object{Hash: k.B58String()})
+	},
+	Marshalers: cmds.MarshalerMap{
+		cmds.Text: func(res cmds.Response) (io.Reader, error) {
+			object := res.Output().(*Object)
+			return strings.NewReader(object.Hash + "\n"), nil
+		},
+	},
+	Type: Object{},
+}
+
+var objectPatchCmd = &cmds.Command{
+	Helptext: cmds.HelpText{
+		Tagline: "Create a new merkledag object based on an existing one",
+		ShortDescription: `
+'ipfs patch <root> [add-link|rm-link] <args>' is a plumbing command used to
```

Comment:
On Mon, Jun 01, 2015 at 07:57:49PM -0700, Juan Batiz-Benet wrote:
> > +'ipfs patch <root> [add-link|rm-link] <args>' is a plumbing command used to
> 
> i'd love to be able to patch the data 
> ```
> echo meow | ipfs patch <root> data

Is that going to be ‘append to Data’ or ‘replace Data entirely’?  I'd
expect the latter.  And this would be more useful if unixfs entries
didn't embed their magic type numbers in Data ;).